powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/AuthenticationService.java

```diff
@@ -551,9 +551,6 @@ public CombinedAuthenticationResponse authenticateCombined(CombinedAuthenticatio
         final IdGeneratorService idGeneratorService = serviceCatalogue.getIdGeneratorService();
 
         final OtpEntity otp = otpService.findOtp(request.getOtpId(), request.getOperationId());
-        if (request.getCredentialName() != null && !request.getCredentialName().equals(otp.getCredentialDefinition().getName())) {
```

Comment:
Can we please check that the OTP has defined credential definition and raise specific error? Otherwise we will get NPE later (it's the bug in the current implementation)